actionpack/lib/abstract_controller/callbacks.rb

```diff
@@ -48,7 +48,8 @@ def _normalize_callback_options(options)
 
       def _normalize_callback_option(options, from, to) # :nodoc:
         if from = options[from]
-          from = Array(from).map {|o| "action_name == '#{o}'"}.join(" || ")
+          _from = Array(from).map(&:to_s)
```

Comment:
Yes, creating sets is slow but we are creating only once and we are calling include on every request so it may be worth using Set here.